include/common/coredump.h

```diff
@@ -0,0 +1,287 @@
+                                       LocalsN); // TODO is this correct?
+
+      Content.push_back(Funcidx);
+      Content.push_back(Codeoffset);
+      //
+      // // TODO get the number type before inserting in the bytes
+      // Is that even required?
+      for (auto &Iter : Locals) {
+        Content.push_back(Iter.unwrap());
+      }
+      for (auto &Iter : Stack) {
+        Content.push_back(Iter.unwrap());
+      }
+    }
+
+    return CoreStack;
+  }
+
+  // AST::GlobalSection collectGlobals(Runtime::StackManager &StackMgr) {
+  //   spdlog::info("Collecting Globals");
+  //   AST::GlobalSection Globals;
+  //   const auto *const CurrentInstance = StackMgr.getModule();
+  //
+  //   auto &GlobalInstances = CurrentInstance->getOwnedGlobalInstances();
+  //   auto &Content = Globals.getContent();
+  //
+  //   for (auto &Iter : GlobalInstances) {
+  //     AST::GlobalSegment Seg;
+  //     auto &Type = Seg.getGlobalType();
+  //     Type = Iter->getGlobalType();
+  //     Content.push_back(Seg);
+  //   }
+  //
+  //   std::cout << Content.size();
+  //
+  //   return Globals;
+  // }
+  // This custom section establishes an index space of modules
+  // that can be referenced in the coreinstances custom section
+  // NOTE This is for multi-module coredumps which are not supported by wasmgdb
+  // as of now
+  // coremodules := customsec(vec(coremodule)) coremodule := 0x0
+  // module-name:name
+  // AST::CustomSection collectCoreModules(Runtime::StackManager &StackMgr) {
+  //   spdlog::info("Constructing Coremodules");
+  //
+  //   AST::CustomSection CoreModules;
+  //   CoreModules.setName("coremodules");
+  //
+  //   auto Frames = StackMgr.getAllFrames();
+  //   std::set<const Runtime::Instance::ModuleInstance *> Names;
+  //
+  //   // get address of all Modules associated to the Frames in the
+  //   StackManager for (const auto &Item : Frames) {
+  //     if (Item.Module == nullptr) {
+  //       continue; // guard
+  //     }
+  //     Names.insert(Item.Module);
+  //   }
+  //
+  //   auto &Content = CoreModules.getContent();
+  //   for (const auto &ModulePtr : Names) {
+  //     const Byte *Bytes = reinterpret_cast<const Byte *>(&ModulePtr);
+  //     Content.insert(Content.end(), Bytes, Bytes + sizeof(ModulePtr));
+  //   }
+  //
+  //   return CoreModules;
+  // }
+  //
+  // coreinstances section
+  // This may not be compatible with current wasmgdb
+  // coreinstances ::= customsec(vec(coreinstance))
+  // coreinstance ::= 0x0 moduleidx:u32 memories:vec(u32) globals:vec(u32)
+  // AST::CustomSection collectCoreInstances(Runtime::StackManager &StackMgr) {
+  //   spdlog::info("Constructing CoreInstances");
+  //
+  //   AST::CustomSection CoreInstances;
+  //   CoreInstances.setName("coreinstances");
+  //   auto Frames = StackMgr.getAllFrames();
+  //
+  //   // struct CoreInstance {
+  //   //   const Runtime::Instance::ModuleInstance Moduleidx;
+  //   //   std::vector<uint8_t> Memories;
```

Comment:
If we simplify the situation, let's say we disable the multi-memory proposal, and you will only have the memories with index 0.